src/main/java/ch/njol/skript/expressions/ExprRandom.java

```diff
@@ -30,62 +34,78 @@
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionList;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.Literal;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.LiteralUtils;
 import ch.njol.util.Kleenean;
 import ch.njol.util.coll.CollectionUtils;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Random")
 @Description("Gets a random item out of a set, e.g. a random player out of all players online.")
-@Examples({"give a diamond to a random player out of all players",
-		"give a random item out of all items to the player"})
+@Examples({
+	"give a diamond to a random player out of all players",
+	"give a random item out of all items to the player"
+})
 @Since("1.4.9")
 public class ExprRandom extends SimpleExpression<Object> {
+
 	static {
 		Skript.registerExpression(ExprRandom.class, Object.class, ExpressionType.COMBINED, "[a] random %*classinfo% [out] of %objects%");
 	}
-	
-	@SuppressWarnings("null")
+
 	private Expression<?> expr;
-	
-	@SuppressWarnings("unchecked")
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		final Expression<?> expr = exprs[1].getConvertedExpression((((Literal<ClassInfo<?>>) exprs[0]).getSingle()).getC());
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (LiteralUtils.hasUnparsedLiteral(exprs[1])) {
+			expr = LiteralUtils.defendExpression(exprs[1]);
+			if (expr instanceof ExpressionList) {
+				Class<?> type = (((Literal<ClassInfo<?>>) exprs[0]).getSingle()).getC();
+				List<Expression<?>> list = Arrays.stream(((ExpressionList<?>) expr).getExpressions())
+						.map(expression -> (Expression<?>) expression.getConvertedExpression(type))
+						.filter(Objects::nonNull)
+						.collect(Collectors.toList());
+				if (list.isEmpty()) {
+					Skript.error("There was no objects of type '" + exprs[0].toString() + "' in the list " + exprs[1].toString());
```

Comment:
```suggestion
					Skript.error("There are no objects of type '" + exprs[0].toString() + "' in the list " + exprs[1].toString());
```
I think this sounds better